zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -1177,6 +1215,12 @@ impl<F: Field> ExecutionConfig<F> {
         transaction: &Transaction,
         call: &Call,
         step: &ExecStep,
+        // Set to true when we're assigning the next step before the current step to have
+        // next step assignments for evaluation of the stored expressions in current step that
+        // depend on the next step.
+        is_next: bool,
```

Comment:
Will `is_final` or `is_final_step` be better? Or `has_next` if `final_step` is not accurate. I understand to use `is` as prefix while it's a `bool` type but in this case `is_next` is not clear to know what it really means.

Reply:
I think `is_final` or `has_next `doesn't hold the real meaning here.  Let me explain why I call this `is_next`.  Each step is assigned two times.  The list of steps is iterated in a window of size 2 where we have (current_step, next_step), and we assign the next first and the current afterwards in each iteration.  See https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/b6be48943f203374ad1d734a52bc66754529de68/zkevm-circuits/src/evm_circuit/execution.rs#L1180-L1206

So if we have the steps [a, b, c] we pair them like (cur=a, next=b), (cur=b, next=c), (cur=c, next=None). And we do the following assignments:
- b (next)
- a (cur)
- b (cur)
- c (next)
- c (cur)

So here the `is_next` means that we're assigning the next step before we assign the current steps for it's "side effects", but we'll go over and assign that step again later.

Do you have another suggestion instead of `is_next` that conveys this meaning?

Reply:
Thanks for your explanation, `is_next` makes sense to me now. :)